drivers/usb/usbstor/scsi.c

```diff
@@ -525,6 +560,46 @@ USBSTOR_SendCBWRequest(
         Context);
 }
 
+static
+NTSTATUS
+USBSTOR_IssueRequestSense(
+    IN PFDO_DEVICE_EXTENSION FDODeviceExtension,
+    IN PIRP Irp,
+    IN PIRP_CONTEXT Context)
+{
+    PIO_STACK_LOCATION IoStack;
+    PSCSI_REQUEST_BLOCK CurrentSrb;
+    PSCSI_REQUEST_BLOCK SenseSrb;
+
+    DPRINT("USBSTOR_IssueRequestSense: \n");
+
+    CurrentSrb = FDODeviceExtension->ActiveSrb;
+    SenseSrb = &Context->SenseSrb;
+    IoStack = IoGetCurrentIrpStackLocation(Irp);
+    IoStack->Parameters.Scsi.Srb = SenseSrb;
+
+    RtlZeroMemory(SenseSrb, sizeof(Context->SenseSrb));
+
+    SenseSrb->Function = SRB_FUNCTION_EXECUTE_SCSI;
+    SenseSrb->Length = sizeof(Context->SenseSrb);
+    SenseSrb->CdbLength = 12;
```

Comment:
`CDB12GENERIC_LENGTH` would be better. Though it would be even nicer to use the right struct (CDB12? Or maybe there's one specific to REQUEST_SENSE)